### PR TITLE
add a new remote workloadmeta only catalog

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

This PR creates a workloadmeta catalog that only contains the remote options. This allows user of this catalog to not include all the dependencies that would be needed to run a local workloadmeta. This also allows to drop the `remotewm` build tag.

The flavor check that used to filter the object received to only containers in the case of the security agent is also removed since most of the CWS logic has been moved to the system-probe anyway.

As a side note, the go deps check uses the build tags defined in the new branch for both the before and after sides, so the diff is not correct here. Most of those dependencies were already not included because of the `remotewm` tag

### Describe how you validated your changes

Code covered by tests. Plus dependency check in CI should be enough

### CODE REVIEW

**Review by KevinFairise2 — APPROVED** [review]

**Review by chouetz — APPROVED** [review]

**Review by gjulianm — APPROVED** [review]

File: comp/core/workloadmeta/collectors/catalog/options.go [hunk 1]
```diff
@@ -3,11 +3,6 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-present Datadog, Inc.
 
-// TODO: (components) Move remote-only to its own catalog, similar to how catalog-less works
```
Comment (gabedos) [inline comp/core/workloadmeta/collectors/catalog/options.go:6]: Could you also remove the similar comment in [catalog.go](https://github.com/DataDog/datadog-agent/blob/d9ab04e0b44a941e112cb3289b9fcbfda10bf634/comp/core/workloadmeta/collectors/catalog/catalog.go#L33)

**Review by gabedos — APPROVED** [review]
LGTM. just small comment change

File: comp/core/workloadmeta/collectors/catalog/options.go (hunk 1 above)
Comment (paulcacheux) [reply comp/core/workloadmeta/collectors/catalog/options.go:6]: ah yes sorry I thought I had removed all of those

**Pull request merged**